src/main/java/com/ldtteam/structurize/management/StructureName.java

```diff
@@ -102,17 +105,18 @@ else if (name.startsWith(Structures.SCHEMATICS_CACHE + Structures.SCHEMATICS_SEP
 
         //The section is the prefix, except fot hut
         section = prefix;
-        if (prefix.equals(Structures.SCHEMATICS_PREFIX))
-        {
-            hut = schematic.split("\\d+")[0];
+        if (prefix.equals(Structures.SCHEMATICS_PREFIX)) {
+            sectionID = schematic.split("\\d+")[0];
             section = Structures.SCHEMATICS_PREFIX;
-            if (ForgeRegistries.BLOCKS.getValue(new ResourceLocation(Constants.MINECOLONIES_MOD_ID, "blockhut" + hut)) != Blocks.AIR || HUTS.contains(hut))
+            for (String origin : StructureLoadingUtils.getOriginMods().keySet())
+                if (ForgeRegistries.BLOCKS.getValue(new ResourceLocation(origin, "blockhut" + sectionID)) instanceof ISectionDefinition)
```

Comment:
I'm not 100% happy about the "blockhut" part of this solution though.

Reply:
Neither am I. I can push it all the way through to remove the blockhut convention, but that means a breaking change for MC, as we would have to change all of the blockhut registered blocks to the new standard, whatever we decided that was. I would probably just go with the building name, but that's my preference.

Reply:
Maybe `HUTS` could be made into a `ResourceLocation` list (or some kind of registry), and then it could be left to MineColonies and other mods to register their hut blocks with Structurize (via an API method, not by directly touching `HUTS`) when they want them to be treated as huts?  Currently it looks like `HUTS` isn't used by anything anyway.

Or probably MineColonies could do it by itself -- it can walk the building registry and get all the hut blocks from all mods, so the other individual mods only need to register a building with MineColonies and don't need to register the hut with Structurize.